### PR TITLE
[#627] Compute Xtext version from Bundle-Version

### PR DESCRIPTION
**Pull request by kthoms**
In development mode Xtext read the version from the 'Maven-Version'
attribute from org.eclipse.xtext.util manifest, which has a static value
of 'unspecified'. This gets replaced in a Gradle build, but does not
work when using the bundle in an Eclipse development environment.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.xtend [hunk 1]
```diff
@@ -72,7 +72,13 @@ class XtextVersion {
 			val url = new URL(Plugin.INSTANCE.baseURL + 'META-INF/MANIFEST.MF')
 			is = url.openStream()
 			val manifest = new Manifest(is)
-			return manifest.getMainAttributes().getValue('Maven-Version')
+			val bundleVersion = manifest.getMainAttributes().getValue('Bundle-Version')
```
Comment (cdietrich) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.xtend]: i am not sure what the reasoning behind this was. and this wont work for nightlys. their bundle version looks like `/Users/dietrich/Downloads/EclipseOxygenXtext214Snapshots.app/Contents/Eclipse/plugins/org.eclipse.xtext_2.14.0.v20180126-1316.jar`

so maybe you can introduce looking at BundleVersion as fallback if Maven version returns unspecified

Comment (cdietrich) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.xtend]: same for release candidates/milestones

**Comment by kthoms** [comment]
Updated. Bundle-Version is only taken as fallback in dev mode.

**Comment by cdietrich** [comment]
please fix failing tests

**Comment by cdietrich** [comment]
@kthoms see my changes

**Comment by kthoms** [comment]
Ah thanks!

**Pull request merged**